paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/CardDetailsUI.kt

```diff
@@ -84,70 +78,85 @@ private fun CardDetailsEditUI(
         ?.collectAsState()
 
     Column {
-        if (billingDetailsForm?.emailElement != null || billingDetailsForm?.phoneElement != null) {
-            ContactInformationSection(billingDetailsForm)
+        val shouldShowNameInContactSection = cardDetailsState == null && billingDetailsForm?.nameElement != null
+        val shouldShowContactSection = billingDetailsForm?.emailElement != null ||
+            billingDetailsForm?.phoneElement != null || shouldShowNameInContactSection
+
+        if (shouldShowContactSection) {
+            ContactInformationSection(
+                billingDetailsForm = billingDetailsForm,
+                includeNameField = shouldShowNameInContactSection
+            )
         }
 
-        Section(
-            title = billingDetailsForm?.let {
-                resolvableString(CoreR.string.stripe_paymentsheet_add_payment_method_card_information)
-            },
-            error = expiryDateState.sectionError()?.resolve(),
-            modifier = Modifier.testTag(UPDATE_PM_CARD_TEST_TAG),
-        ) {
-            Column {
-                if (billingDetailsForm?.nameElement != null) {
-                    SectionElementUI(
-                        enabled = true,
-                        element = SectionElement.wrap(sectionFieldElement = billingDetailsForm.nameElement),
-                        hiddenIdentifiers = emptySet(),
-                        lastTextFieldIdentifier = null
+        if (cardDetailsState != null) {
+            Section(
+                title = billingDetailsForm?.let {
+                    resolvableString(CoreR.string.stripe_paymentsheet_add_payment_method_card_information)
+                },
+                error = cardDetailsState.expiryDateState.sectionError()?.resolve(),
+                modifier = Modifier.testTag(UPDATE_PM_CARD_TEST_TAG),
+            ) {
+                Column {
+                    // Only show name in card section if there's no contact info section to put it in
+                    val shouldShowNameInCardSection = billingDetailsForm?.nameElement != null &&
+                        (billingDetailsForm.emailElement == null && billingDetailsForm.phoneElement == null)
```

Comment:
We should keep the existing behavior for cards, where the name is always shown in the card section.

Reply:
that's how it works right now, I think the comment is outdated
<img width="389" alt="image" src="https://github.com/user-attachments/assets/e6941605-38d2-483b-ae8b-3e4c396f8d6e" />